pkg/plugin/util/util_test.go

```diff
@@ -101,4 +101,60 @@ var _ = Describe("Cover plugin util helpers", func() {
 			Expect(lines).To(Equal([]string{"noemptylines"}))
 		})
 	})
+
+	Describe("HasFileContentWith", Ordered, func() {
+		const (
+			path    = "testdata/PROJECT"
+			content = `# Code generated by tool. DO NOT EDIT.
+# This file is used to track the info used to scaffold your project
+# and allow the plugins properly work.
+# More info: https://book.kubebuilder.io/reference/project-config.html
+domain: example.org
+layout:
+- go.kubebuilder.io/v4
+- helm.kubebuilder.io/v1-alpha
+plugins:
+  helm.kubebuilder.io/v1-alpha: {}
+repo: github.com/example/repo
+version: "3"
+`
+		)
+
+		BeforeAll(func() {
+			err := os.MkdirAll("testdata", 0o755)
+			Expect(err).NotTo(HaveOccurred())
+
+			if _, err = os.Stat(path); os.IsNotExist(err) {
+				err = os.WriteFile(path, []byte(content), 0o644)
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
+
+		AfterAll(func() {
+			err := os.RemoveAll("testdata")
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should return true when file contains the expected content", func() {
+			content := "repo: github.com/example/repo"
+			found, err := HasFileContentWith(path, content)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(found).To(BeTrue())
+		})
+
+		It("should return true when file contains multiline expected content", func() {
+			content := `plugins:
+  helm.kubebuilder.io/v1-alpha: {}`
+			found, err := HasFileContentWith(path, content)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(found).To(BeTrue())
+		})
+
+		It("should return false when file does not contain the expected content", func() {
+			content := "nonExistentContent"
+			found, err := HasFileContentWith(path, content)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(found).To(BeFalse())
+		})
+	})
 })
```

Comment:
Also, we should not add ANY change in the plugin/util/util_test
We do not need to create any new func here.